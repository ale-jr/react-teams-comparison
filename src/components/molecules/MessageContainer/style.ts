import styled from 'styled-components'


export const MessageContainer = styled.div`
    display: flex;
    align-items: center;
    justify-content: center;
    flex-direction: column;
    gap: ${({ theme }) => theme.spacing.md};

    height: 100%;

    p{
        font-size: ${({theme})=>theme.typography.size.xl};
    }

`