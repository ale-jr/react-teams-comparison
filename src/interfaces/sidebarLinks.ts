import { IconDefinition } from "@fortawesome/fontawesome-svg-core"
export interface SidebarLink {
    icon?: IconDefinition
    title: string
    to: string
    links?: SidebarLink[]
}