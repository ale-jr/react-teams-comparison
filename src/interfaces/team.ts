export interface TeamValues {
  participation: number;
  eNPS: number;
  overallEngagement: number;
  recognition: number;
  feedback: number;
  relationshipWithPeers: number;
  relationshipWithManager: number;
  satisfaction: number;
  alignment: number;
  happiness: number;
  wellness: number;
  personalGrowth: number;
  ambassadorship: number;
}

export interface Team extends TeamValues {
  name: string;
}
