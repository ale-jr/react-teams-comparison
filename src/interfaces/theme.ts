export interface ColorVariants {
  darkest?: string;
  dark?: string;
  default: string;
  light: string;
  lightest?: string;
  contrast: string;
  "0"?: string;
  "50"?: string;
  "100"?: string;
  "200"?: string;
  "300"?: string;
  "400"?: string;
  "500"?: string;
  "600"?: string;
}

export type ColorVariantKeys = keyof ColorVariants;

export interface Sizing {
  xs: string;
  sm: string;
  md: string;
  lg: string;
  xl: string;
}

export type SizingKeys = keyof Sizing;

export interface Typography {
  size: Sizing;
  weight: {
    regular: string;
    semibold: string;
    bold: string;
  };
  font: {
    heading: string;
    body: string;
  };
}

export interface Colors {
  neutral: ColorVariants;
  primary: ColorVariants;
  success: ColorVariants;
  danger: ColorVariants;
}
export type ColorKeys = keyof Colors;

export interface Animation {
  duration: string;
  timingFunction: string;
  shorthand: string;
}

export interface Theme {
  colors: Colors;
  spacing: Sizing;
  typography: Typography;
  animation: Animation;
}
