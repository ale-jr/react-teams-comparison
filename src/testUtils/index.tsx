import { render, RenderOptions, RenderResult } from '@testing-library/react'
import ThemeProvider from 'providers/ThemeProvider'
import { ReactElement } from 'react'


const Providers: React.FC<{
    children: JSX.Element | JSX.Element[]
}> = ({ children }) => <ThemeProvider>{children}</ThemeProvider>


const customRender = (ui: ReactElement, options?: Omit<RenderOptions, "queries">): RenderResult => render(ui, {
    wrapper: Providers,
    ...options
})

export * from '@testing-library/react'
export {customRender as render}