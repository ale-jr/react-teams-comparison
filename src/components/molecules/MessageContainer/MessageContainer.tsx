import {
  faSpinner,
  faExclamationTriangle,
} from "@fortawesome/free-solid-svg-icons";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { MessageContainer as BaseMessageContainer } from "./style";

interface MessageContainerProps {
  type: "error" | "loading";
  message: string;
}

const MessageContainer: React.FC<MessageContainerProps> = ({
  type,
  message,
}) => {
  return (
    <BaseMessageContainer>
      {type === "error" && (
        <FontAwesomeIcon icon={faExclamationTriangle} size="5x" />
      )}
      {type === "loading" && (
        <FontAwesomeIcon icon={faSpinner} spin size="5x" />
      )}
      <p>{message}</p>
    </BaseMessageContainer>
  );
};
export default MessageContainer;
