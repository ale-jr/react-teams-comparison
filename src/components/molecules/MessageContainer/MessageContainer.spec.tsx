import { Team } from "interfaces/team";
import "jest-styled-components";
import { render } from "testUtils";
import MessageContainer from "./index";

describe("MessageContainer", () => {
  it("should match snapshot", () => {
    const errorComponent = render(
      <MessageContainer message="Oh no!" type="error" />
    );
    expect(errorComponent.asFragment()).toMatchSnapshot();

    const loadingContainer = render(
      <MessageContainer message="Fiding the best job ever!" type="loading" />
    );
    expect(loadingContainer.asFragment()).toMatchSnapshot();
  });
});
