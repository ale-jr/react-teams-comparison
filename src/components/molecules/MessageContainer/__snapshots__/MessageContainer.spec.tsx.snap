// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageContainer should match snapshot 1`] = `
<DocumentFragment>
  .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 1em;
  height: 100%;
}

.c0 p {
  font-size: 1.5em;
}

<div
    class="c0"
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-triangle-exclamation fa-5x "
      data-icon="triangle-exclamation"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M506.3 417l-213.3-364c-16.33-28-57.54-28-73.98 0l-213.2 364C-10.59 444.9 9.849 480 42.74 480h426.6C502.1 480 522.6 445 506.3 417zM232 168c0-13.25 10.75-24 24-24S280 154.8 280 168v128c0 13.25-10.75 24-23.1 24S232 309.3 232 296V168zM256 416c-17.36 0-31.44-14.08-31.44-31.44c0-17.36 14.07-31.44 31.44-31.44s31.44 14.08 31.44 31.44C287.4 401.9 273.4 416 256 416z"
        fill="currentColor"
      />
    </svg>
    <p>
      Oh no!
    </p>
  </div>
</DocumentFragment>
`;

exports[`MessageContainer should match snapshot 2`] = `
<DocumentFragment>
  .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 1em;
  height: 100%;
}

.c0 p {
  font-size: 1.5em;
}

<div
    class="c0"
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-spinner fa-spin fa-5x "
      data-icon="spinner"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M304 48C304 74.51 282.5 96 256 96C229.5 96 208 74.51 208 48C208 21.49 229.5 0 256 0C282.5 0 304 21.49 304 48zM304 464C304 490.5 282.5 512 256 512C229.5 512 208 490.5 208 464C208 437.5 229.5 416 256 416C282.5 416 304 437.5 304 464zM0 256C0 229.5 21.49 208 48 208C74.51 208 96 229.5 96 256C96 282.5 74.51 304 48 304C21.49 304 0 282.5 0 256zM512 256C512 282.5 490.5 304 464 304C437.5 304 416 282.5 416 256C416 229.5 437.5 208 464 208C490.5 208 512 229.5 512 256zM74.98 437C56.23 418.3 56.23 387.9 74.98 369.1C93.73 350.4 124.1 350.4 142.9 369.1C161.6 387.9 161.6 418.3 142.9 437C124.1 455.8 93.73 455.8 74.98 437V437zM142.9 142.9C124.1 161.6 93.73 161.6 74.98 142.9C56.24 124.1 56.24 93.73 74.98 74.98C93.73 56.23 124.1 56.23 142.9 74.98C161.6 93.73 161.6 124.1 142.9 142.9zM369.1 369.1C387.9 350.4 418.3 350.4 437 369.1C455.8 387.9 455.8 418.3 437 437C418.3 455.8 387.9 455.8 369.1 437C350.4 418.3 350.4 387.9 369.1 369.1V369.1z"
        fill="currentColor"
      />
    </svg>
    <p>
      Fiding the best job ever!
    </p>
  </div>
</DocumentFragment>
`;
